Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm

```diff
@@ -53,17 +59,76 @@
 static NSString * const worldKey = @"world";
 
 static NSString * const  cssKey = @"css";
-// FIXME: <https://webkit.org/b/261765> Consider adding support for cssOrigin.
 
 static NSString * const mainWorld = @"MAIN";
 static NSString * const isolatedWorld = @"ISOLATED";
 
-static int64_t const mainFrameID = 0;
-
+// FIXME: <https://webkit.org/b/261765> Consider adding support for cssOrigin.
 // FIXME: <https://webkit.org/b/261765> Consider adding support for injectImmediately.
 
 namespace WebKit {
 
+static inline void parseTargetInjectionOptions(NSDictionary *targetInfo, WebExtensionScriptInjectionParameters& parameters, NSString **outExceptionString)
+{
+    NSNumber *tabID = targetInfo[tabIDKey];
+    std::optional<WebExtensionTabIdentifier> identifier = toWebExtensionTabIdentifier(tabID.doubleValue);
+    if (!identifier) {
+        *outExceptionString = toErrorString(nil, tabIDKey, @"'%@' is not a tab identifier", tabID);
+        return;
+    }
+
+    parameters.tabIdentifier = identifier;
+
+    if (NSArray *frameIDs = targetInfo[frameIDsKey]) {
+        Vector<WebExtensionFrameIdentifier> frames;
+        for (NSNumber *frameID in frameIDs) {
+            auto identifier = toWebExtensionFrameIdentifier(frameID.doubleValue);
+            if (!identifier) {
+                *outExceptionString = toErrorString(nil, frameIDsKey, @"'%@' is not a frame identifier", frameID);
+                return;
+            }
+
+            frames.append(identifier.value());
+        }
+
+        parameters.frameIDs = frames;
+    }
+
+    if (!boolForKey(targetInfo, allFramesKey, false))
+        parameters.frameIDs = Vector<WebExtensionFrameIdentifier> { WebExtensionFrameConstants::MainFrameIdentifier };
```

Comment:
I think `... = Vector { WebExtensionFrameConstants::MainFrameIdentifier };` will work, since the template can be inferred by the type in the `{ }`.